### PR TITLE
codeblocksFull: 20.03 -> unstable-2023-01-17

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

This commit bumps codeblocksFull to drop most patches and to build against wxWidgets 3.2.

https://forums.codeblocks.org/index.php/topic,3299.0.html

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/codeblocks/default.nix [hunk 1]
```diff
@@ -1,81 +1,91 @@
...
 }:
 
-with lib;
-
 stdenv.mkDerivation rec {
-  name = "${pname}-${lib.optionalString contribPlugins "full-"}${version}";
-  version = "20.03";
   pname = "codeblocks";
+  version = "unstable-2022-11-02";
+  name = "${pname}-${lib.optionalString contribPlugins "full-"}${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/codeblocks/default.nix:24]: Please do not overwrite name and split this into name and version

File: pkgs/applications/editors/codeblocks/default.nix [hunk 2]
```diff
@@ -1,81 +1,91 @@
...
 
-  src = fetchurl {
-    url = "mirror://sourceforge/codeblocks/Sources/${version}/codeblocks-${version}.tar.xz";
-    sha256 = "1idaksw1vacmm83krxh5zlb12kad3dkz9ixh70glw1gaibib7vhm";
+  src = fetchFromGitHub {
+    owner = "arnholm";
+    repo = "codeblocks_sfmirror";
+    rev = "4d3041fe5201173effe1bda2c8db274f74ff47b2";
+    leaveDotGit = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/codeblocks/default.nix:30]: Why do we need this? Can we patch instead?

Comment (wegank) [reply pkgs/applications/editors/codeblocks/default.nix:30]: Arch Linux was already applying [13 patches](https://github.com/archlinux/svntogit-community/blob/packages/codeblocks/trunk/PKGBUILD), maybe more for darwin...

**Comment by wegank** [comment]
Fix for aarch64-linux in https://github.com/NixOS/nixpkgs/commit/9e9b992a5b318579a82e3d60cbdbf9108fffb5fa ?

**Comment by wegank** [comment]
@ofborg build codeblocksFull codeblocksFull.passthru.tests

File: pkgs/applications/editors/codeblocks/default.nix (hunk 2 above)
Comment (Mindavi) [reply pkgs/applications/editors/codeblocks/default.nix:30]: LeaveDotGit will result in an unreproducible hash

**Pull request closed without merging**